### PR TITLE
Update developer readme formatting and https links

### PR DESCRIPTION
**Pull request by jodygarnett**
This is a non functional change based on feedback from 2.20-RC testing

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/README.txt [hunk 1]
```diff
@@ -4,76 +4,92 @@ GeoServer Development requires Java 8, Maven, and git. Due to subtle changes in
...
 
 1) Install Java Runtime Environment
-Download and install Java 8 runtime environment, as provided by AdoptOpenJDK windows installers.
-https://adoptopenjdk.net
 
-Update Windows *Environment Variables*:
+   Download and install Java 8 runtime environment, as provided by Adoptium windows installers:
+   https://adoptium.net
 
+   Update Windows *Environment Variables*:
```
Comment (bradh) [inline src/README.txt:46]: I think the MSI offers to do this for you. Its probably OK as is, and I will provide a follow-up PR if appropriate after further testing.

File: src/README.txt [hunk 2]
```diff
@@ -4,76 +4,92 @@ GeoServer Development requires Java 8, Maven, and git. Due to subtle changes in
...
-== Troubleshooting ==
+Troubleshooting
+---------------
 
 The build process may fail because of several reasons:
 
 * Unavailable dependencies - Maven tries to download dependencies which might not be available on the server side yet.
-Solution: Try again in some minutes.
+
+  Solution: Try again in some minutes.
```
Comment (bradh) [inline src/README.txt:86]:
```suggestion
  Solution: Try again in a few minutes.

```

File: src/release/developer_README.txt [hunk 3]
```diff
@@ -4,78 +4,92 @@ GeoServer Development requires Java 8, Maven, and git. Due to subtle changes in
...
+---------------
 
 The build process may fail because of several reasons:
 
 * Unavailable dependencies - Maven tries to download dependencies which might not be available on the server side yet.
-Solution: Try again in some minutes.
 
-* Failing tests - Maven runs existing tests automatically. If some of them fail, the build fails.
-Solution: You can tell Maven not to run the tests. This is discouraged.
+  Solution: Try again in some minutes.
```
Comment (bradh) [inline src/release/developer_README.txt:86]:
```suggestion
  Solution: Try again in a few minutes.

```

**Review by bradh — APPROVED** [review]
A couple of minor suggestions, unrelated to the main goal, but probably worth fixing at the same time.

File: src/README.txt [hunk 4]
```diff
@@ -4,76 +4,92 @@ GeoServer Development requires Java 8, Maven, and git. Due to subtle changes in
...
 
 * Failing tests - Maven runs existing tests automatically. If some of them fail, the build fails.
-Solution: You can tell maven not to run the tests. This is discouraged.
-Bug the developers instead or fix the test and send a patch, thanks!
-If you really just want to disable the test, run maven like so:
-  mvn -DskipTests=true install
+
+  Solution: You can tell maven not to run the tests. This is discouraged.
```
Comment (mprins) [inline src/README.txt:90]:
```suggestion
  Solution: You can tell Maven not to run the tests. This is discouraged.

```

File: src/release/developer_README.txt [hunk 5]
```diff
@@ -4,78 +4,92 @@ GeoServer Development requires Java 8, Maven, and git. Due to subtle changes in
...
 
-Bug the developers instead or fix the test and send a patch, thanks!
-If you really just want to disable the test, run Maven like so:
+* Failing tests - Maven runs existing tests automatically. If some of them fail, the build fails.
 
-  mvn -DskipTests=true install
+  Solution: You can tell maven not to run the tests. This is discouraged.
+  Bug the developers instead or fix the test and send a patch, thanks!
+  
+  If you really just want to disable the test, run maven like so:
```
Comment (mprins) [inline src/release/developer_README.txt:93]:
```suggestion
  If you really just want to disable the test, run Maven like so:

```

**Review by bradh — APPROVED** [review]

**Comment by jodygarnett** [comment]
Thanks @mprins, we should back port and adjust as needed also.

**Pull request merged**